x-pack/test/functional/apps/ml/memory_usage/memory_usage_page.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+export default function ({ getService }: FtrProviderContext) {
+  const ml = getService('ml');
+  const esArchiver = getService('esArchiver');
+
+  const jobId = 'sample_job';
+
+  describe('ML memory usage page', function () {
+    this.tags(['ml']);
+
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/farequote');
+
+      const jobConfig = ml.commonConfig.getADFqSingleMetricJobConfig(jobId);
+
+      // Create and open AD job
+      await ml.api.createAnomalyDetectionJob(jobConfig);
+      await ml.api.openAnomalyDetectionJob(jobId);
+
+      await ml.navigation.navigateToMl();
+      await ml.navigation.navigateToMemoryUsage();
+    });
+
+    after(async () => {
+      await ml.api.closeAnomalyDetectionJob(jobId);
+      await ml.api.cleanMlIndices();
+    });
+
+    it('opens page with nodes tab selected', async () => {
+      await ml.memoryUsage.assertMemoryUsageTabIsSelected('nodes');
+    });
+
+    it('allows sorting', async () => {
+      await ml.memoryUsage.sortColumn('tableHeaderCell_name_1');
+      await ml.memoryUsage.assertColumnIsSorted('tableHeaderCell_name_1', 'descending');
+    });
+
+    it('allows searching for a node', async () => {
+      await ml.memoryUsage.searchForNode('ftr');
+      await ml.memoryUsage.assertRowCount(1);
+    });
+
+    it('expands node details and displays memory usage details', async () => {
+      await ml.memoryUsage.expandRow();
+      await ml.memoryUsage.assertNodeExpandedDetailsPanelsExist();
+      await ml.memoryUsage.selectNodeExpandedRowTab('mlNodesOverviewPanelMemoryTab');
+      await ml.memoryUsage.assertChartItemsSelectedByDefault();
+      await ml.memoryUsage.assertTreeChartExists();
+    });
+
+    it('clears selected chart items', async () => {
+      await ml.memoryUsage.clearSelectedChartItems();
+      await ml.memoryUsage.assertEmptyTreeChartExists();
+    });
+
+    it('selects memory usage tab and displays chart', async () => {
+      await ml.memoryUsage.selectTab('memory-usage');
+      await ml.memoryUsage.assertTreeChartExists();
+
+      await ml.memoryUsage.clearSelectedChartItems();
```

Comment:
Is it worth adding a step where you leave just the Anomaly Detection Job item and assert the tree chart exists, before clearing all?

Reply:
Hmm, I'm not sure what you mean exactly.
This test is responsible for ensuring that the chart is displayed properly in the `memory usage` tab. 
![image](https://github.com/user-attachments/assets/dffa7a30-1bdf-4c16-9171-90965d052994)
Then, it clears the combo box and checks if the chart displays an empty state.
The same thing was done earlier, but for a chart inside an expanded row.

Reply:
Don't worry. I was thinking we could do a click on the DFA and Trained model items above the chart to get it into this state, but probably not worth the effort as there is only 1 AD job in this case.

<img width="892" alt="Screenshot 2024-10-25 at 10 07 55" src="https://github.com/user-attachments/assets/7bd7ee62-04ac-4576-b574-32463cf18098">